composer.json

```diff
@@ -38,7 +38,8 @@
         "phpstan/phpstan":              "^0.11.4",
         "phpstan/phpstan-phpunit":      "^0.11",
         "infection/infection":          "^0.12.2",
-        "symfony/console":              "^4.2.4"
+        "symfony/console":              "^4.2.4",
+        "vimeo/psalm": "^3.3"
```

Comment:
Note: I just committed whatever I had stashed locally, so I'd just need a rough high-level review of this feature.